### PR TITLE
feat: add bitvector module

### LINKED ISSUE #390: Bit Vector utils
We need a module that allows us to perform some operations on bitstrings in a similar manner to how they are performed in spec bitvectors. That is:

- Allow little-endian bit indexing.
- Querying if a single bit is set.
- Querying if multiple bits are set.
- Shifting bits right or left.
- Allowing a vector of size N to be built from a bigger bitstring or bytestring.
- Being able to convert from an integer in binary notation (e.g. 0b0101).

### PR DESCRIPTION
**Pull request by Arkenan**
Closes #390

### CODE REVIEW

File: lib/utils/bit_vector.ex [hunk 1]
```diff
@@ -0,0 +1,94 @@
+defmodule LambdaEthereumConsensus.Utils.BitVector do
+  @moduledoc """
+  Set of utilities to interact with bit vectors, represented as bitstrings.
+  The vector is indexed with little endian bit notation. That is, the bit 0
+  is the less significant 1.
```
Comment (MegaRedHand) [inline lib/utils/bit_vector.ex:5]: > That is, the bit 0 is the less significant 1.

I suggest rephrasing it to:

> That is, the 0th bit is the least significant bit of the corresponding byte.

**Review by MegaRedHand — COMMENTED** [review]
Nice job!

File: test/unit/bit_vector_test.exs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+defmodule BitVectorTest do
```
[the remark this answers is not in the document]
Comment (Arkenan) [reply test/unit/bit_vector_test.exs:1]: Done for 24 bit vectors

File: lib/utils/bit_vector.ex (hunk 1 above)
Comment (Arkenan) [reply lib/utils/bit_vector.ex]: Applied

File: lib/utils/bit_vector.ex [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  """
+  @spec set?(t, non_neg_integer) :: boolean
+  def set?(bit_vector, index) do
+    skip = bit_size(bit_vector) - index - 1
+    match?(<<_::size(skip), 1::size(1), _::bitstring>>, bit_vector)
+  end
+
+  @doc """
+  True if all bits in the specified range are set to 1.
+  Equivalent to all?(bit_vector[first..last]) in the specs.
```
[the remark this answers is not in the document]
Comment (Arkenan) [reply lib/utils/bit_vector.ex]: Applied.

**Review by MegaRedHand — APPROVED** [review]

**Pull request merged**